presto-tests/src/test/java/com/facebook/presto/tests/TestSqlFunctions.java

```diff
@@ -251,6 +251,31 @@ public void testTemporarySqlFunctions()
         assertQuery(createSessionWithTempFunctionFoo(), "SELECT foo(foo(2))", "SELECT 8");
     }
 
+    @Test
+    public void testShowTemporaryFunctions()
+    {
+        MaterializedResult result = computeActual(createSessionWithTempFunctionFoo(), "SHOW FUNCTIONS");
+        MaterializedRow row = result.getMaterializedRows().get(result.getMaterializedRows().size() - 1);
+        assertEquals(row.getField(0), "foo");
+    }
+
+    @Test
+    public void testShowCreateTemporaryFunction()
+    {
+        MaterializedRow result = computeActual(createSessionWithTempFunctionFoo(), "SHOW CREATE FUNCTION foo(bigint)").getMaterializedRows().get(0);
+        String createFunctionFooFormatted = "CREATE FUNCTION foo (\n" +
```

Comment:
The result of the `SHOW CREATE foo` should be a `CREATE TEMPORARY FUNCTION` statement. The result SQL by itself is not semantically valid, as it is attempting to create a permanent function `foo` without specifying catalog and schema.

Reply:
done